File: cpp/alerts/tracker_reply.cpp
#include "./tracker_reply.h"

namespace ltrs {

int tracker_reply_alert_get_num_peers(lt::tracker_reply_alert* alert) {
    return alert->num_peers;
}

uint8_t tracker_reply_alert_get_version(lt::tracker_reply_alert* alert) {
    return static_cast<uint8_t>(alert->version);
}

} // namespace ltrs

File: src/ffi/alerts/tracker_reply.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type tracker_reply_alert = crate::ffi::ffi::tracker_reply_alert;
    }

    unsafe extern "C++" {
        include!("cpp/alerts/tracker_reply.h");
        unsafe fn tracker_reply_alert_get_num_peers(alert: *mut tracker_reply_alert) -> i32;
        unsafe fn tracker_reply_alert_get_version(alert: *mut tracker_reply_alert) -> u8;
    }
}
